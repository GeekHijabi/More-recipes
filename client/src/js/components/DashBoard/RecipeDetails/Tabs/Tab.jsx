import React from 'react';

const Tab = props => (
  <a
    className={
      props.isActive ?
        'nav-item nav-link active' :
        'nav-item nav-link'}
    id="ingredients-tab"
    data-toggle="tab"
    role="tab"
    aria-controls="ingredients"
    aria-selected="false"
    onClick={props.handleClick}
    href={`#${props.tabOption}`}
  >
    {props.tabOption.name}
  </a>
);

export default Tab;
