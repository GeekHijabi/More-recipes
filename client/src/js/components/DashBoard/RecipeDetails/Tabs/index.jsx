import React from 'react';
import Tab from './Tab';

const tabOptions = [
  { id: 1, name: 'Ingredients', isActive: false },
  { id: 2, name: 'Descriptions', isActive: true },
  { id: 3, name: 'Reviews', isActive: false }
];

const Tabs = props => (
  <nav
    className="nav nav-tabs white"
    id="myTab"
    role="tablist"
  >
    {
      tabOptions.map(tab => (
        <Tab
          tabOption={tab}
          key={tab.id}
          isActive={(props.activeTab === tab[0])}
          handleClick={() => props.changeTab(tab)}
        />
      ))
    }
    {props.isActive}
  </nav>
);

export default Tabs;
