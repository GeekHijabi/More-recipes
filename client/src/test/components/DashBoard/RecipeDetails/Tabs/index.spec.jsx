import React from 'react';
import { shallow } from 'enzyme';
import Tabs
  from '../../../../../js/components/DashBoard/RecipeDetails/Tabs/index';

const props = {
  isActive: false,
  activeTab: {
    name: 'ingredients'
  }
};


describe('Tabs snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Tabs {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

// describe('handleClick', () => {
//   it('should change the tab on click', () => {
//     const wrapper = shallow(<Tabs {...props} />);
//     wrapper.instance().handleClick();
//     wrapper.simulate('click', activeTab);
//     expect(wrapper.instance().props.changeTab()).toBeCalled();
//   });
// });

