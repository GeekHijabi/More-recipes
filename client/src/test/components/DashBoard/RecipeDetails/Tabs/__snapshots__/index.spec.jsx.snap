// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs snapshot should render correctly 1`] = `
<nav
  className="nav nav-tabs white"
  id="myTab"
>
  <Tab
    handleClick={[Function]}
    isActive={false}
    key="1"
    tabOption={
      Object {
        "id": 1,
        "isActive": true,
        "name": "Ingredients",
      }
    }
  />
  <Tab
    handleClick={[Function]}
    isActive={false}
    key="2"
    tabOption={
      Object {
        "id": 2,
        "isActive": false,
        "name": "Descriptions",
      }
    }
  />
  <Tab
    handleClick={[Function]}
    isActive={false}
    key="3"
    tabOption={
      Object {
        "id": 3,
        "isActive": false,
        "name": "Reviews",
      }
    }
  />
</nav>
`;
